precompile/results/predicate_results.go

```diff
@@ -0,0 +1,112 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package results
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/codec/linearcodec"
+	"github.com/ava-labs/avalanchego/utils/units"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	Version        = uint16(0)
+	MaxResultsSize = units.MiB
+)
+
+var Codec codec.Manager
+
+func init() {
+	Codec = codec.NewManager(MaxResultsSize)
+
+	c := linearcodec.NewDefault()
+	errs := wrappers.Errs{}
+	errs.Add(
+		c.RegisterType(PredicateResults{}),
+		Codec.RegisterCodec(Version, c),
+	)
+	if errs.Errored() {
+		panic(errs.Err)
+	}
+}
+
+// TxPredicateResults is a map of results for each precompile address to the resulting byte array.
+type TxPredicateResults map[common.Address][]byte
+
+// PredicateResults encodes the precompile predicate results included in a block on a per transaction basis.
+// PredicateResults is not thread-safe.
```

Comment:
don't we need this to be thread-safe?

Reply:
We currently only modify this synchronously in the worker and during block verification. Is there a specific case you're concerned about?